docs/proposals/20220725-managed-kubernetes.md

```diff
@@ -0,0 +1,521 @@
+So that I can eliminate the responsibility of owning and SREing the Control Plane from the Cluster service consumer and cluster admin.
+
+### Current State of Managed Kubernetes in CAPI
+
+#### EKS in CAPA
+
+- https://cluster-api-aws.sigs.k8s.io/topics/eks/index.html
+- Feature Status: GA
+- CRDs
+  - AWSManagedControlPlane - provision EKS cluster
+  - AWSManagedMachinePool - corresponds to EKS managed node pool
+- Supported Flavors
+  - AWSManagedControlPlane with MachineDeployment / AWSMachine
+  - AWSManagedControlPlane with MachinePool / AWSMachinePool
+  - AWSManagedControlPlane with MachinePool / AWSManagedMachinePool
+- Bootstrap Provider
+  - Cluster API bootstrap provider EKS (CABPE)
+* Features
+  - Provisioning/managing an Amazon EKS Cluster
+  - Upgrading the Kubernetes version of the EKS Cluster
+  - Attaching self-managed machines as nodes to the EKS cluster
+  - Creating a machine pool and attaching it to the EKS cluster (experimental)
+  - Creating a managed machine pool and attaching it to the EKS cluster
+  - Managing “EKS Addons”
+  - Creating an EKS fargate profile (experimental)
+  - Managing aws-iam-authenticator configuration
+
+#### AKS in CAPZ
+
+- https://capz.sigs.k8s.io/topics/managedcluster.html
+- Feature Status: Experimental
+- CRDs
+  - AzureManagedControlPlane, AzureManagedCluster - provision AKS cluster
+  - AzureManagedMachinePool - corresponds to AKS node pool
+- Supported Flavor
+  - AzureManagedControlPlane + AzureManagedCluster with AzureManagedMachinePool
+
+#### OKE in CAPOCI
+
+- https://github.com/oracle/cluster-api-provider-oci/issues/110
+- Design discussion starting
+
+### Managed Kubernetes API Design Approaches
+
+When discussing the different approaches to represent a managed Kubernetes service in CAPI, we will be using the implementation of GKE support in CAPG as an example, as this isn’t currently implemented.
+
+> NOTE: “naming things is hard” so the names of the kinds/structs/fields used in the CAPG examples below are illustrative only and are not the focus of this proposal. There is debate, for example, as to whether `GCPManagedCluster` or `GKECluster` should be used. This type of discussion will be within the CAPG proposal.
+
+The following section discusses different API implementation options along with pros and cons of each.
+
+#### Option 1: Single kind for Control Plane and Infrastructure
+
+This option introduces a new single resource kind:
+
+- **GCPManagedControlPlane**: this represents both a control-plane (i.e. GKE) and infrastructure required for the cluster. It contains properties for both the general cloud infrastructure (that would traditionally be represented by an infrastructure cluster) and the managed Kubernetes control plane (that would traditionally be represented by a control plane provider).
+
+```go
+type GCPManagedControlPlaneSpec struct {
+// Project is the name of the project to deploy the cluster to.
```

Comment:
nit: indent

(similar below in other code blocks)

Reply:
All code now indented.